pkg/controller/config/config_controller.go

```diff
@@ -197,13 +217,35 @@ func (r *ReconcileConfig) Reconcile(ctx context.Context, request reconcile.Reque
 
 	r.cacheManager.ExcludeProcesses(newExcluder)
 	configSourceKey := aggregator.Key{Source: "config", ID: request.NamespacedName.String()}
+	// if gvksToSync is empty, cachemanager will CancelData for stale expectations
 	if err := r.cacheManager.UpsertSource(ctx, configSourceKey, gvksToSync); err != nil {
+		// for readiness, try to cancel the expectations on error for the gvks
```

Comment:
Let's say a syncsource has 2 gvks:
- gvkValid
- gvkInvalid

When the watch for gvkInvalid fails, we will cancel expectations for both gvks?

This also doesn't work well with SyncSets b/c we could cancel a data expectation that is also part of a SyncSet.